src/main/java/bio/terra/pdao/bigquery/BigQueryProject.java

```diff
@@ -51,6 +55,14 @@ public BigQuery getBigQuery() {
         return bigQuery;
     }
 
+    // TODO: REVIEWERS PLEASE CHECK: Should these methods be in here? On the one hand, it is convenient. But it
+    // mixes the duties of this class: it is supplying both the cache and the BQ methods. Unfortunately, it
+    // doesn't supply all of the BQ methods, so sometimes getBigQuery is needed. Seems like it could be
+    // improved.
+    //
+    // In general, BigQueryPdao is too big and needs refactoring. Perhaps when that is done, we can also
+    // re-think this code structure.
```

Comment:
I thought about this as well -- these convenience methods do DRY up some code but it feels incomplete because a lot of the time you need to just fetch the BigQuery instance and use that directly. I think we should leave it for now and revisit when we reconsider the BigQueryPdao..

One idea: 

1. isolate the code that operates on the BigQuery instance into one component
2. have one component responsible for taking in DrDatasets and turning them into BqDatasets using the component # 1 from above
3. have (possibly another or the # 2) component responsible for taking in DataSnapshots and turning them into BqDatasets using # 1
4. have this "factory" generate (# 1s) or BigQueryProjects that # 1 accepts to operate on

Created [DR-488](https://broadworkbench.atlassian.net/browse/DR-488) to track